### PR TITLE
redfang: init at 2.5

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
#81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126371` at c9b82f3f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>redfang</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/tools/networking/redfang/default.nix:15:15:
```
   |
15 |   patches = [ ./makefile-install.diff ];
   |               ^
```
</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 126371` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>redfang</li>
  </ul>
</details>

File: pkgs/tools/networking/redfang/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  NIX_CFLAGS_COMPILE = "-Wno-format-security";
+
+  buildInputs = [ bluez ];
+
+  meta = with lib; {
+    description = "A small proof-of-concept application to find non discoverable bluetooth devices";
+    homepage = "https://gitlab.com/kalilinux/packages/redfang";
+    license = licenses.unfree; # Upstream has no license
```
Comment (fabaff) [inline pkgs/tools/networking/redfang/default.nix:26]: License seems to be [`GPL2`](https://gitlab.com/kalilinux/packages/redfang/-/blob/kali/master/fang.c#L7-9).

File: pkgs/tools/networking/redfang/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitLab {
+    group = "kalilinux";
+    owner = "packages";
+    repo = pname;
+    rev = "upstream/${version}";
+    sha256 = "sha256-dF9QmBckyHAZ+JbLr0jTmp0eMu947unJqjrTMsJAfIE=";
+  };
+
+  patches = [ ./makefile-install.diff ];
```
Comment (fabaff) [inline pkgs/tools/networking/redfang/default.nix:15]: Can you try to push that upstream?

I doubt that there will be a new release but other distributions may need the patch as well.

**Review by fabaff — COMMENTED** [review]
- [x] package built (see https://github.com/NixOS/nixpkgs/pull/126371#issuecomment-858352234)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-126371]$ sudo ./results/redfang/bin/fang 
redfang - the bluetooth hunter ver 2.5
(c)2003 @stake Inc
author:   Ollie Whitehouse <ollie@atstake.com>
enhanced: threads by Simon Halsall <s.halsall@eris.qinetiq.com>
enhanced: device info discovery by Stephen Kapp <skapp@atstake.com>
Scanning 281474976710656 address(es)
Address range 00:00:00:00:00:00 -> ff:ff:ff:ff:ff:ff
^Cne 2 - 00:00:00:00:00:02 
```

File: pkgs/tools/networking/redfang/default.nix (hunk 2 above)
Comment (fortuneteller2k) [reply pkgs/tools/networking/redfang/default.nix]: Submitted a merge request here: https://gitlab.com/kalilinux/packages/redfang/-/merge_requests/1

**Pull request merged**